src/Riok.Mapperly/Descriptors/MappingBuilder/ObjectPropertyMappingBuilderContext.cs

```diff
@@ -24,60 +24,54 @@ public class ObjectPropertyMappingBuilderContext
 {
     private readonly Dictionary<PropertyPath, PropertyNullDelegateAssignmentMapping> _nullDelegateMappings = new();
     private readonly HashSet<string> _unmappedSourcePropertyNames;
+    private readonly IReadOnlyCollection<string> _ignoredUnmatchedTargetPropertyNames;
+    private readonly IReadOnlyCollection<string> _ignoredUnmatchedSourcePropertyNames;
 
     public ObjectPropertyMappingBuilderContext(MappingBuilderContext builderContext, ObjectPropertyMapping mapping)
     {
         BuilderContext = builderContext;
         Mapping = mapping;
 
-        IgnoredTargetPropertyNames = GetIgnoredTargetProperties();
-        _unmappedSourcePropertyNames = GetSourcePropertyNames(IgnoredTargetPropertyNames);
+        // build ignored property names
+        var ignoredTargetPropertyNames = GetIgnoredTargetProperties();
+        IgnoredSourcePropertyNames = GetIgnoredSourceProperties();
+
+        // build unmapped source property names
```

Comment:
This comment could be improved. Right now, it does not have a lot of "value". For example

> Start building the unmapped source property names. Initialize with all source property names, then remove the ones that are mapped/ignored.

The same applies to the other comments in this constructor. Because of the good naming of the variables etc., the comments are quite redundant at the moment. I would expand the comments to explain "why" something happens, not "what" happens.

If you can't think of something to improve the comments, then it would be better to remove them IMO. Or add a single, big comment at the start of the constructor, explaining the general concept of what is happening

Reply:
Removed all comments since they don't add any value and couldn't think of any better comments since the code seems pretty much self explaining.